plutus-tx/src/PlutusTx/Builtins.hs

```diff
@@ -374,7 +374,11 @@ encodeUtf8 :: BuiltinString -> BuiltinByteString
 encodeUtf8 = BI.encodeUtf8
 
 matchList :: forall a r . BI.BuiltinList a -> r -> (a -> BI.BuiltinList a -> r) -> r
-matchList l nilCase consCase = BI.chooseList l (const nilCase) (\_ -> consCase (BI.head l) (BI.tail l)) ()
+-- For some reason, everything here needs to be non-strict, otherwise we get the
```

Comment:
Um, wtf?

Reply:
Not about INLINABLE. Does INLINABLE even matter, if we have `-fno-omit-interface-pragmas`? I guess not?

Reply:
Not about INLINABLE. Does INLINABLE even matter, if we have `-fno-omit-interface-pragmas`? I guess not?

Reply:
Yes it matters, it can affect whether GHC adds an unfolding and how much it optimizes it.